homeassistant/components/keba/binary_sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Support for KEBA charging station binary sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station binary sensors")
+    keba = hass.data[DOMAIN]
+
+    sensors = [
+        KebaBinarySensor(keba, 'Online', 'Wallbox', 'connectifity'),
+        KebaBinarySensor(keba, 'Plug', 'Plug', 'plug'),
+        KebaBinarySensor(keba, 'State', 'Charging state', 'power'),
+        KebaBinarySensor(keba, 'Tmo FS', 'Failsafe Mode', 'safety')
+    ]
+    async_add_entities(sensors)
+
+
+class KebaBinarySensor(BinarySensorDevice):
+    """Representation of a binary sensor of a KEBA charging station."""
+
+    def __init__(self, keba, key, sensor_name, device_class):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = sensor_name
+        self._device_class = device_class
+        self._is_on = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Deactivate polling. Data updated by KebaHandler."""
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return self._keba.device_name + '_' + self._name
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._is_on
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the binary sensor."""
+        return self._attributes.items()
+
+    async def async_update(self):
+        """Get latest cached states from the device."""
+        if self._key == 'Online':
+            self._is_on = self._keba.get_value(self._key)
+            self._attributes['Product'] = self._keba.get_value('Product')
+            self._attributes['Serial'] = self._keba.get_value('Serial')
+            self._attributes['Firmware'] = self._keba.get_value('Firmware')
+            self._attributes['Device uptime'] = \
```

Comment:
What kind of time is this? We only allow absolute utc timestamps in state attributes. No relative times. Relative times cause the state to update just because time passes.

Reply:
Actually it is a relative time (uptime of the device in seconds). It might be usefull to see if there was a restart of the charging station but I think it is not an important number for automation. So i will remove it for this PR. If it turns out that it is useful it could be added later on.